### PR TITLE
add starting edge architecture

### PR DESCRIPTION
**Pull request by Gkrumbach07**
closes: https://github.com/opendatahub-io/odh-dashboard/issues/2247

## Description

- added context component
- added hook to get all models
- added utils to organize and fetch certain parts of a model

## Test Impact

added util tests

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by Gkrumbach07** [comment]
Merging this. 

_note: skipping reviews as this feature is a special case_

File: frontend/src/concepts/edge/useEdgeModels.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { listK8sPipelineRunsByLabel } from '~/api';
+import useFetchState, { FetchStateCallbackPromise } from '~/utilities/useFetchState';
+import { EdgeModel } from './types';
+import { organizePipelineRuns } from './utils';
+
+export const useEdgeModels = (namespace: string) => {
+  const callback = React.useCallback<FetchStateCallbackPromise<EdgeModel[]>>(async () => {
+    const runs = await listK8sPipelineRunsByLabel(
```
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply frontend/src/concepts/edge/useEdgeModels.ts:9]: yeah. it is specifically for models. which is not a true k8s resource. it its underlaying resource is pipelines

[the remark this answers is not in the document]
Comment (dpanshug) [reply frontend/src/concepts/edge/useEdgeModels.ts:9]: Oh I see

File: frontend/src/api/k8s/pipelineRuns.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+): Promise<PipelineRunKind[]> =>
+  k8sListResource<PipelineRunKind>({
+    model: PipelineRunModel,
+    queryOptions: {
+      ns: namespace,
+      queryParams: { labelSelector: label },
+    },
+  }).then((listResource) => listResource.items);
+
+export const getK8sPipelineRun = async (
```
Comment (dpanshug) [inline frontend/src/api/k8s/pipelineRuns.ts:17]: Isn't this should be `listK8sPipelineRun`? 

Should I change this in my working PR?

File: frontend/src/api/k8s/pipelineRuns.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  namespace: string,
+  name: string,
+): Promise<PipelineRunKind[]> =>
+  k8sListResource<PipelineRunKind>({
+    model: PipelineRunModel,
+    queryOptions: {
+      ns: namespace,
+      queryParams: { name: name },
```
Comment (dpanshug) [inline frontend/src/api/k8s/pipelineRuns.ts:25]: Also, what's the need of name query, if we are fetching the whole list?

File: frontend/src/api/k8s/pipelineRuns.ts (hunk 2 above)
Comment (Gkrumbach07) [reply frontend/src/api/k8s/pipelineRuns.ts:17]: yeah go ahead. i wasnt using it . i just made it for an example

File: frontend/src/api/k8s/pipelineRuns.ts (hunk 3 above)
Comment (Gkrumbach07) [reply frontend/src/api/k8s/pipelineRuns.ts:25]: yeah i was not using this function, just made it for an example

**Pull request merged**